### PR TITLE
JetStream3: add BigInt tests

### PR DESCRIPTION
**Pull request by shvaikalesh**
#### d4714544b5ca9101493ffa0b07ca7e9bfad7d687
<pre>
JetStream3: add BigInt tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=247574">https://bugs.webkit.org/show_bug.cgi?id=247574</a>
&lt;rdar://102045614&gt;

Reviewed by Yusuke Suzuki.

Adds 5 tests that are heavy on arithmetic operations on BigInt:
bigint-noble-bls12-381, bigint-noble-secp256k1, bigint-noble-ed25519,
bigint-paillier, and bigint-bigdenary.

* PerformanceTests/JetStream3/JetStreamDriver.js:
(Driver.prototype.async fetchResources):
* PerformanceTests/JetStream3/bigint/bigdenary-benchmark.js: Added.
* PerformanceTests/JetStream3/bigint/bigdenary-bundle.js: Added.
* PerformanceTests/JetStream3/bigint/noble-benchmark.js: Added.
* PerformanceTests/JetStream3/bigint/noble-bls12-381-bundle.js: Added.
* PerformanceTests/JetStream3/bigint/noble-ed25519-bundle.js: Added.
* PerformanceTests/JetStream3/bigint/noble-secp256k1-bundle.js: Added.
* PerformanceTests/JetStream3/bigint/paillier-benchmark.js: Added.
* PerformanceTests/JetStream3/bigint/paillier-bundle.js: Added.
* PerformanceTests/JetStream3/bigint/web-crypto-sham.js: Added.
* PerformanceTests/JetStream3/in-depth.html:

Canonical link: <a href="https://commits.webkit.org/257085@main">https://commits.webkit.org/257085@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/e8749dd4df6a68dd5705bb7bc408d89addc88362

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97865 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/7098 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/31043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/107348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7537 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/35872 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/90242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/103985 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103507 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5668 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/84488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/90242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/87539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/31043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/90242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/1081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/31043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/1068 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/31043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/5897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/84488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2416 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2335 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/31043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | |

### CODE REVIEW

**Review by Constellation — APPROVED** [review]
rs=me

File: PerformanceTests/JetStream3/in-depth.html [hunk 1]
```diff
@@ -655,6 +655,50 @@ <h3>
          Source code: <a href="SunSpider/3d-cube.js">3d-cube.js</a>
         </dd>
 
+        <dt id="bigint-noble-bls12-381">bigint-noble-bls12-381</dt>
```
Comment (shvaikalesh) [inline PerformanceTests/JetStream3/in-depth.html:658]: _cc_ @paulmillr: adding your stuff so we could optimize `BigInt` based on real-world usage.

Comment (paulmillr) [reply PerformanceTests/JetStream3/in-depth.html:658]: @shvaikalesh Sounds awesome. Is there an approximate date for official release of jetstream 3?

Comment (shvaikalesh) [reply PerformanceTests/JetStream3/in-depth.html:658]: @paulmillr Per 🍎 policy I can't comment on future releases, even approximately (and I don't even know), yet adding tests for newish JS features is my top priority and I can see light at the end of the tunnel. Also, JavaScriptCore team constantly optimizes newest JS additions without regard to official releases of any benchmark.

**Pull request merged**